mountpoint-s3/src/fs.rs

```diff
@@ -536,17 +537,18 @@ where
         ino: InodeNo,
         fh: u64,
         offset: i64,
-        data: &[u8],
+        data: ChecksummedSlice<'_>,
```

Comment:
I know we're doing it this way because we want to "do checksums as early as possible", but I think we might want to push the checksum calculation down into the `Uploader`. It's a very small difference in time (a couple of lock acquisitions?) and therefore risk, and moving it will make it much easier to disable checksums or allow different checksum algorithms in the future. I think moving it also saves you the need for `ChecksummedSlice` at all -- you can just maintain a running `Hasher`.

In the future you could imagine actually doing this in `fuser` instead, as the bytes are being read off the FUSE device, and then the entire interface would have this checksummed type.